app/code/Magento/Cms/Model/Wysiwyg/Images/PrepareImage.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Cms\Model\Wysiwyg\Images;
+
+use Magento\Catalog\Helper\Data;
+use Magento\Cms\Helper\Wysiwyg\Images as ImagesHelper;
+
+class PrepareImage
+{
+    /**
+     * @var ImagesHelper
+     */
+    private $imagesHelper;
+
+    /**
+     * @var Data
+     */
+    private $catalogHelper;
+
+    /**
+     * PrepareImage constructor.
+     * @param ImagesHelper $imagesHelper
+     * @param Data $catalogHelper
+     */
+    public function __construct(
+        ImagesHelper $imagesHelper,
+        Data $catalogHelper
+    ) {
+        $this->imagesHelper = $imagesHelper;
+        $this->catalogHelper = $catalogHelper;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public function execute(array $data): string
+    {
+        $filename = $this->imagesHelper->idDecode($data['filename']);
+        $storeId = (int)$data['store_id'];
+
+        $this->catalogHelper->setStoreId($storeId);
```

Comment:
Can you please check why this call to catalog helper is needed.

Reply:
@sivaschenko , did a check on different store view with and without that call and seems to not change anything on the rendered text in the content. I just included it on the model to not miss anything out from the original class. Let me know your thoughts.